community/kernel/src/main/java/org/neo4j/kernel/api/txtracking/TransactionIdTracker.java

```diff
@@ -39,9 +40,11 @@
     private static final TimeUnit POLL_UNIT = TimeUnit.MILLISECONDS;
 
     private final TransactionIdStore transactionIdStore;
+    private final LongPredicate isAvailable;
 
-    public TransactionIdTracker( TransactionIdStore transactionIdStore )
+    public TransactionIdTracker( TransactionIdStore transactionIdStore, LongPredicate isAvailable )
```

Comment:
I think `LongPredicate` is a bit too generic here. Can we maybe instead pass in `AvailabilityGuard` and then use it's `#isAvailable(long)` method? This would result in same functionality but with more descriptive and concrete types.